gitindex/index.go

```diff
@@ -403,34 +391,31 @@ func IndexGitRepo(opts Options) error {
 		if when := commit.Committer.When; when.After(opts.BuildOptions.RepositoryDescription.LatestCommitDate) {
 			opts.BuildOptions.RepositoryDescription.LatestCommitDate = when
 		}
+	}
 
-		tree, err := commit.Tree()
-		if err != nil {
-			return fmt.Errorf("commit.Tree: %w", err)
-		}
+	if opts.Incremental && opts.BuildOptions.IncrementalSkipIndexing() {
```

Comment:
we seem to be able to do this sooner now. i am guessing this is just an optimization you noticed and doesn't change behaviour?

Reply:
Yes, this doesn't  seem to change existing behavior after I stared at it for a while. The code that came before this call never modified any of the build options, so the result of `opts.Incremental && opts.BuildOptions.IncrementalSkipIndexing()` shouldn't change.